lms/djangoapps/grades/tasks.py

```diff
@@ -24,9 +28,27 @@
 
 @task(default_retry_delay=30, routing_key=settings.RECALCULATE_GRADES_ROUTING_KEY)
 def recalculate_subsection_grade(
+        # pylint: disable=unused-argument
         user_id, course_id, usage_id, only_if_higher, weighted_earned, weighted_possible, **kwargs
 ):
     """
+    Shim to allow us to modify this task's signature without blowing up production on deployment.
+    """
+    recalculate_subsection_grade_v2.apply(
+        kwargs=dict(
+            user_id=user_id,
+            course_id=course_id,
+            usage_id=usage_id,
+            only_if_higher=only_if_higher,
+            expected_modified_time=kwargs.get('expected_modified_time', 0),  # Use the unix epoch as a default
+            score_deleted=kwargs['score_deleted'],
+        )
+    )
+
+
+@task(default_retry_delay=30, routing_key=settings.RECALCULATE_GRADES_ROUTING_KEY)
+def recalculate_subsection_grade_v2(**kwargs):
```

Comment:
Old -> New: New needs to handle the case where the parameter doesn't exist.
New -> Old: New needs to handle the case where the parameter won't be honored.

We version the task so that the latter does not happen.